File: clients/python/examples/collections-find-and-rerank-vectorize-lexical.py
from astrapy import DataAPIClient

# Get an existing collection
client = DataAPIClient()
database = client.get_database(
    "**API_ENDPOINT**", token="**APPLICATION_TOKEN**"
)
collection = database.get_collection("**COLLECTION_NAME**")

# Find documents
cursor = collection.find_and_rerank(
    sort={
        "$hybrid": {
            "$vectorize": "A tree in the woods",
            "$lexical": "house hill grassy",
        },
    },
)

# Iterate over the found documents
for result in cursor:
    print(result.document)

File: clients/typescript/examples/collections-find-and-rerank-vectorize-lexical.ts
import { DataAPIClient } from "@datastax/astra-db-ts";

// Get an existing collection
const client = new DataAPIClient();
const database = client.db("**API_ENDPOINT**", {
  token: "**APPLICATION_TOKEN**",
});
const collection = database.collection("**COLLECTION_NAME**");

(async function () {
  // Find documents
  const cursor = collection.findAndRerank(
    {},
    {
      sort: {
        $hybrid: {
          $vectorize: "A tree in the woods",
          $lexical: "house hill grassy",
        },
      },
    },
  );

  // Iterate over the found documents
  for await (const result of cursor) {
    console.log(result.document);
  }
})();
